app/src/main/java/com/ranveeraggarwal/letrack/activities/AddPersonActivity.java

```diff
@@ -37,24 +48,48 @@
     String selectedFrequency;
     String selectedStartDate;
 
+    int REQUEST_CONTACTS = 1;
+    String TAG = "AddPersonActivity";
+    boolean hasContactPermission = false;
+
     DatabaseAdapter databaseAdapter;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+
         setContentView(R.layout.activity_add_person);
+        if (ContextCompat.checkSelfPermission(
+                this, Manifest.permission.READ_CONTACTS) != PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(this, new String[] {Manifest.permission.READ_CONTACTS}, REQUEST_CONTACTS);
+        } else {
+            hasContactPermission = true;
+        }
 
         databaseAdapter = new DatabaseAdapter(this);
 
         toolbar = (Toolbar) findViewById(R.id.add_app_bar);
         setSupportActionBar(toolbar);
         assert getSupportActionBar() != null;
-        getSupportActionBar().setHomeButtonEnabled(true);
         getSupportActionBar().setDisplayHomeAsUpEnabled(true);
         getSupportActionBar().setTitle("Add Person");
 
         nameField = (TextView) findViewById(R.id.name_field);
 
+        chooseFromContacts = (TextView) findViewById(R.id.choose_from_contacts);
+        chooseFromContacts.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                addNameOnClickListener();
+            }
+        });
+
+        if (hasContactPermission) {
+            chooseFromContacts.setVisibility(View.VISIBLE);
+        } else {
+            chooseFromContacts.setVisibility(View.GONE);
+        }
```

Comment:
Isn't this redundant given you already have the `updateContactButton` function? You can call it here.